src/observable/subscribe.js

```diff
@@ -34,13 +34,23 @@ SubscribeObserver.prototype.event = function (t, x) {
 SubscribeObserver.prototype.end = function (t, x) {
   if (!this.disposable.disposed) {
     var s = this.subscriber
-    doDispose(this.fatalError, s, s.complete, s.error, this.disposable, x)
+    var fatalError = this.fatalError
+    Promise.resolve(this.disposable.dispose()).then(function () {
+      if (typeof s.complete === 'function') {
+        s.complete(x)
+      }
+    }).catch(function (e) {
+      throwError(e, s.error, fatalError)
```

Comment:
Since a subscriber is an object, I think we should always call `s.error` with `s` as the `thisArg`.  I'm thinking we could pass `s` here instead of `s.error`.

Also, it looks like `throwError` is only call with `fatalError` as it's 3rd parameter (and it's hard to imagine calling it with anything else, in the current setup).  So, it may make sense just to remove the parameter and call `fatalError` directly in throwError.

Maybe something like:

```js
function throwError (e1, subscriber) {
  if (typeof subscriber.error !== 'function') {
    try {
      subscriber.error(e1)
    } catch (e2) {
      fatalError(e2)
    }
  } else {
    fatalError(e1)
  }
}